libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/util/SalesforceAnalyticsLogger.java

```diff
@@ -0,0 +1,222 @@
+        log(context, SalesforceLogger.Level.ERROR, tag, message, e);
+    }
+
+    /**
+     * Logs a warning log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     */
+    public static void w(Context context, String tag, String message) {
+        log(context, SalesforceLogger.Level.WARN, tag, message);
+    }
+
+    /**
+     * Logs a warning log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     * @param e Exception to be logged.
+     */
+    public static void w(Context context, String tag, String message, Throwable e) {
+        log(context, SalesforceLogger.Level.WARN, tag, message, e);
+    }
+
+    /**
+     * Logs an info log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     */
+    public static void i(Context context, String tag, String message) {
+        log(context, SalesforceLogger.Level.INFO, tag, message);
+    }
+
+    /**
+     * Logs an info log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     * @param e Exception to be logged.
+     */
+    public static void i(Context context, String tag, String message, Throwable e) {
+        log(context, SalesforceLogger.Level.INFO, tag, message, e);
+    }
+
+    /**
+     * Logs a debug log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     */
+    public static void d(Context context, String tag, String message) {
+        log(context, SalesforceLogger.Level.DEBUG, tag, message);
+    }
+
+    /**
+     * Logs a debug log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     * @param e Exception to be logged.
+     */
+    public static void d(Context context, String tag, String message, Throwable e) {
+        log(context, SalesforceLogger.Level.DEBUG, tag, message, e);
+    }
+
+    /**
+     * Logs a verbose log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     */
+    public static void v(Context context, String tag, String message) {
+        log(context, SalesforceLogger.Level.VERBOSE, tag, message);
+    }
+
+    /**
+     * Logs a verbose log line.
+     *
+     * @param context Context.
+     * @param tag Log tag.
+     * @param message Log message.
+     * @param e Exception to be logged.
+     */
+    public static void v(Context context, String tag, String message, Throwable e) {
+        log(context, SalesforceLogger.Level.VERBOSE, tag, message, e);
+    }
+
+    private static void log(Context context, SalesforceLogger.Level level, String tag, String message) {
+        if (context != null) {
+            SalesforceLogger.getLogger(COMPONENT_NAME, context).log(level, tag, message);
```

Comment:
Nope - it shouldn't. I did consider that carefully when adding those calls.
- Our library will throw a `RuntimeException` if any code is executed before `SalesforceSDKManager.init...` has been called. This is by design, because most parts of our library operate under the assumption that an instance of `SalesforceSDKManager` (and thereby an `appContext`) will be available.
- None of the tests use this logging framework for that very reason (to avoid such a situation).